source/crud/crud.rst

```diff
@@ -884,6 +884,17 @@ Basic
      */
     collation: Optional<Document>;
 
+    /**
+     * The index to use.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * This option is only supported by servers >= 4.2. Older servers >= 3.4 will report an error for using this option.
+     * For servers < 3.4, the driver MUST raise an error if the caller explicitly provides a value.
```

Comment:
3.4 is significant, as that was the first server version where `update` started rejecting unknown options. As such, I think we can rely on a 3.4+ server to report an error if the option is unsupported and raise a client-side error for older server versions where `hint` might be otherwise ignored. Similar to `collation`, silently ignoring a `hint` option could be dangerous in that more or less documents could match than intended (consider the case of a sparse index).

Reply:
The wire version to check for here would be 5+ (see: [Server Wire Version and Feature List](https://github.com/mongodb/specifications/blob/master/source/wireversion-featurelist.rst)).
Most all drivers do version checks via wire protocol version, but specs vary by referring to wire and/or server versions. The wording here is internally consistent with the CRUD spec.

Side note: Despite the fact that the server does not guarantee to bump its wire version with each release (past results are not indicative of future... yada yada), this has been the case thus far.